component/azstorage/block_blob.go

```diff
@@ -572,24 +572,38 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 
 	// For some directories 0 byte meta file may not exists so just create a map to figure out such directories
 	var dirList = make(map[string]bool)
-
+	var errors error = nil
 	for _, blobInfo := range listBlob.Segment.BlobItems {
-		attr := &internal.ObjAttr{
-			Path:   split(bb.Config.prefixPath, *blobInfo.Name),
-			Name:   filepath.Base(*blobInfo.Name),
-			Size:   *blobInfo.Properties.ContentLength,
-			Mode:   0,
-			Mtime:  *blobInfo.Properties.LastModified,
-			Atime:  dereferenceTime(blobInfo.Properties.LastAccessedOn, *blobInfo.Properties.LastModified),
-			Ctime:  *blobInfo.Properties.LastModified,
-			Crtime: dereferenceTime(blobInfo.Properties.CreationTime, *blobInfo.Properties.LastModified),
-			Flags:  internal.NewFileBitMap(),
-			MD5:    blobInfo.Properties.ContentMD5,
+		attr := &internal.ObjAttr{}
+		if blobInfo.Properties.CustomerProvidedKeySHA256 != nil && *blobInfo.Properties.CustomerProvidedKeySHA256 != "" {
+			log.Trace("BlockBlob::List : blob is encrypted with customer provided key so fetching metadata explicitly using REST")
+			attr, err = bb.getAttrUsingRest(*blobInfo.Name)
+			if err != nil {
+				log.Err("BlockBlob::List : Failed to get properties of blob %s", *blobInfo.Name)
+				if errors == nil {
+					errors = err
+				} else {
+					errors = fmt.Errorf("error fetching metadata for blob: %s, %w; %w", *blobInfo.Name, errors, err)
+				}
```

Comment:
What is the use of this? We can directly return the error after this loop.

Reply:
Aren't you updating errors variable with the last returned err?
My suggestion would be to log the error here (if not nil) and return err after this loop.

Reply:
Maybe we can just log and return the first error we got here. If GetProperties fails for a blob, we should not continue.
In this case, you can remove the errors variable and lines 615-617.